target/linux/ath79/dts/ar9344_devolo_dlan-pro-500-wp.dts

```diff
@@ -0,0 +1,230 @@
+		};
+
+		led_status_green: status_green {
+			label = "green:status";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+
+		led_eth_green: eth_green {
+			label = "green:eth";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		led_dlan: dlan_greed {
+			label = "green:dlan";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		wifi {
+			label = "WIFI button";
+			linux,code = <KEY_RFKILL>;
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		dlan {
+			label = "DLAN button";
+			linux,code = <BTN_0>;
+			gpios = <&gpio 21 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	ahb: ahb {
+		// Overwrite the register for GMAC0 connection
+		gmac: gmac@18070000 {
+			compatible = "qca,ar9340-gmac";
+			// 0b010001 = 0x11
+			//    |   +- RGMII_GMAC0
+			//    +-> MII_GMAC0_SLAVE
+			reg = <0x18070000 0x11>;
+		};
+	};
+};
+
+&gpio {
+	plc_power {
+		gpio-hog;
+		line-name = "PLC power";
+		gpios = <&gpio 13 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot: partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x40000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				label = "Config1";
+				reg = <0x50000 0x10000>;
+				read-only;
+			};
+
+			partition@60000 {
+				label = "Config2";
+				reg = <0x60000 0x10000>;
+				read-only;
+			};
+
+			partition@70000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x70000 0x780000>;
+			};
+
+			art: partition@7f0000 {
+				label = "art";
+				reg = <0x7f0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&ref {
+	clock-frequency = <40000000>;
+};
+
+// GMAC0 is connected to a AR7400 PLC in PHY mode
+&eth0 {
+	status = "okay";
+
+	// This is <PLL_1000 PLL_100 PLL_10>
+	pll-data = <0x0e000000 0x00000101 0x00001616>;
+
+	nvmem-cells = <&macaddr_art_1002>;
+	nvmem-cell-names = "mac-address";
+	mac-address-increment = <3>;
+
+	phy-mode = "rgmii";
+
+	gmac-config {
+		device = <&gmac>;
+		rgmii-gmac0 = <1>;
+	};
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+// GMAC1 is connected to the internal switch
+&eth1 {
+	status = "okay";
+
+	pll-data = <0x16000000 0x00000101 0x00001616>;
+
+	nvmem-cells = <&macaddr_art_1002>;
+	nvmem-cell-names = "mac-address";
+	mac-address-increment = <1>;
+
+	phy-mode = "gmii";
+	phy-handle = <&phy1>;
+};
+
+&wmac {
+	status = "okay";
+
+	mtd-cal-data = <&art 0x1000>;
+};
+
+&pcie {
+	status = "okay";
+
+	wifi@0,0 {
+		compatible = "qcom,ath10k";
```

Comment:
Wiki says [ath9k](https://openwrt.org/toh/devolo/dlan_pro_wireless_500_plus)?

Reply:
Yeah, I copied this over from other device trees. And I couldn't find any ath9k examples. If `"qcom,ath9k"` is valid, I can change.

Reply:
I've searched the device trees and found `compatible = "pci168c,0030";` in other ar934x that have `ath9k: wifi...` declarations.

I can give this a try if preferred.

Reply:
Don't have the coding skill to judge if it only has cosmetic or structural implications, just made a comment to keep this in my github notification. Just noticed the word 'ath10k' in a quick scan of your sourcecode when looking for a codebase for similar ath9k hardware [D-Link DHP-1565](https://openwrt.org/toh/d-link/dhp-1565) that may benefit from your code.

Reply:
Is the wifi@0,0 entry needed at all? This is a PCIe device which gets probed over the PCIe bus and we only add the strange reg property in device tree.

Reply:
Thanks! I've removed the whole `&pcie` block. I confirmed that WiFi is still working.

I've also updated the default network config to enable the switch for the LAN ports.